src/engine.ts

```diff
@@ -768,6 +811,93 @@ export class MLCEngine implements MLCEngineInterface {
     return JSON.stringify(function_list);
   }
 
+  /**
+   * Given a string outputMessage, parse it as a JSON object and return an array of tool calls.
+   *
+   * Expect outputMessage to be a valid JSON string, and expect it to be an array of Function with
+   * fields `arguments` and `name`.
+   */
+  private getToolCallFromOutputMessage(
+    outputMessage: string,
+    isStreaming: boolean,
+  ):
+    | Array<ChatCompletionMessageToolCall>
+    | Array<ChatCompletionChunk.Choice.Delta.ToolCall> {
```

Comment:
We should consider make two separate function headers like how `chatCompletion` does since currently TypeScript won't be able to discriminate the return type correctly.